cmd/thriftrw-list-deps/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+
+	"github.com/jessevdk/go-flags"
+
+	"go.uber.org/thriftrw/compile"
+)
+
+type args struct {
+	ThriftFile string `positional-arg-name:"file" description:"Path to the Thrift file"`
+}
+
+var opts struct {
+	RelativeTo string `long:"relative-to" description:"If specified, output paths will be relative to this directory"`
+	Args       args   `positional-args:"yes" required:"yes"`
+}
+
+// listDependentThrifts lists the Thrift files that the given Thrift file depends on or includes, both directly (through imports) and
+// indirectly (through transitive imports).
+//
+// The returned file paths are absolute, unless relativeDir parameter is given, in which case the paths are relative to
+// the relativeDir.
+func listDependentThrifts(inputThriftFile string, relativeDir string) ([]string, error) {
+	dependentThriftFiles := make([]string, 0)
+
+	module, err := compile.Compile(inputThriftFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not compile %q: %v", inputThriftFile, err)
+	}
+
+	for _, included := range module.Includes {
```

Comment:
This won't cover transitive imports. You'll need to use [`module.Walk`](https://godoc.org/go.uber.org/thriftrw/compile#Module.Walk) for that.

```go
module.Walk(func(included *compile.Module) error {
    ...
})
```